src/Bridge/Doctrine/Orm/QueryAwarePartialPaginatorInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Doctrine\Orm;
+
+use ApiPlatform\Core\DataProvider\PartialPaginatorInterface;
+use Doctrine\ORM\Query;
+
+interface QueryAwarePartialPaginatorInterface extends PartialPaginatorInterface
```

Comment:
Why `Partial` if it is used in the `Paginator`?

Reply:
the name was suggested by @soyuka. Not sure here.

Reply:
I think it's because the method was on the `AbstractPaginator` before. @soyuka do you confirm `QueryAwarePaginatorInterface` would be better?

Reply:
I think the `Partial` part of the name is redundant, even though I understand why @soyuka suggested that.

Reply:
It should not extend `PartialPaginatorInterface`.

Reply:
Perhaps we could just make it `QueryAwareInterface`. It could be used in other places as necessary, not tied to the `Paginator`.

Reply:
I like QueryAwarePartialPaginatorInterface if you indeed extend the AbstractPaginator, at least we know what we talk about by looking at the name no?

Reply:
The ability to get the `Query` object is not tied to the `Paginator` though... It's just a general interface that gives access to an underlying `Query` object.

Reply:
@soyuka @teohhanhui @alanpoulain so what do you decide about this?

Reply:
Let's go with a generic `QueryAwareInterface`.